### PR TITLE
chore: Clean up design tokens

### PR DESCRIPTION
**Pull request by codecaaron**
## Platform Theme Separation - Part 3
- [Part 1](https://github.com/Codecademy/client-modules/pull/1754)
- [Part 2](https://github.com/Codecademy/client-modules/pull/1755)
- [Part 3](https://github.com/Codecademy/client-modules/pull/1756)

Monolith PR: https://github.com/codecademy-engineering/Codecademy/pull/24393

#### Token clean up
- Many tokens have been cleaned up as they are no longer used or serve no purpose.  (Typography, BoxShadows)
- Unused utilities like responsive helpers can just be used with `polished` or achieved with system props.
- Removes some unnecessary tagged template stuff causing legibility problems with class names from standalone styles.
- Removes docs areas that referenced these scales.
- Adds direct number references for font weights so they are less opaque.

### PR Checklist

- [x] Related to designs: [Figma](https://www.figma.com/file/QplOE1d1rHbQLu0PtvsWLX/Colors-%E2%9C%85?node-id=487%3A2)
- [x] Related to JIRA ticket: https://codecademy.atlassian.net/browse/GM-174 https://codecademy.atlassian.net/browse/GM-228
- [x] I have run this code to verify it works
- [x] This PR includes unit tests for the code change

### CODE REVIEW

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-icons@4.12.3-alpha.2519d8.0<br/>@codecademy/gamut-illustrations@0.14.4-alpha.2519d8.0<br/>@codecademy/gamut-kit@0.3.101-alpha.2519d8.0<br/>@codecademy/gamut-labs@16.0.6-alpha.2519d8.0<br/>@codecademy/gamut-patterns@0.1.4-alpha.2519d8.0<br/>@codecademy/gamut-styles@10.0.1-alpha.2519d8.0<br/>@codecademy/gamut-tests@2.3.31-alpha.2519d8.0<br/>@codecademy/gamut@32.18.8-alpha.2519d8.0<br/>@codecademy/konamimojisplosion@0.2.31-alpha.2519d8.0<br/>@codecademy/markdown-overrides@0.5.43-alpha.2519d8.0<br/>@codecademy/styleguide@37.5.9-alpha.2519d8.0<br/>@codecademy/variance@0.11.3-alpha.2519d8.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://609de8e3cd15b148f2ccfabd--gamut-preview.netlify.app>https://609de8e3cd15b148f2ccfabd--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/609de8e3cd15b148f2ccfabd>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-icons@4.12.3-alpha.cdff67.0<br/>@codecademy/gamut-illustrations@0.14.4-alpha.cdff67.0<br/>@codecademy/gamut-kit@0.3.101-alpha.cdff67.0<br/>@codecademy/gamut-labs@16.0.6-alpha.cdff67.0<br/>@codecademy/gamut-patterns@0.1.4-alpha.cdff67.0<br/>@codecademy/gamut-styles@10.0.1-alpha.cdff67.0<br/>@codecademy/gamut-tests@2.3.31-alpha.cdff67.0<br/>@codecademy/gamut@32.18.8-alpha.cdff67.0<br/>@codecademy/konamimojisplosion@0.2.31-alpha.cdff67.0<br/>@codecademy/markdown-overrides@0.5.43-alpha.cdff67.0<br/>@codecademy/styleguide@37.5.9-alpha.cdff67.0<br/>@codecademy/variance@0.11.3-alpha.cdff67.0<br/></pre>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-icons@4.12.3-alpha.502a81.0<br/>@codecademy/gamut-illustrations@0.14.4-alpha.502a81.0<br/>@codecademy/gamut-kit@0.3.101-alpha.502a81.0<br/>@codecademy/gamut-labs@16.0.6-alpha.502a81.0<br/>@codecademy/gamut-patterns@0.1.4-alpha.502a81.0<br/>@codecademy/gamut-styles@10.0.1-alpha.502a81.0<br/>@codecademy/gamut-tests@2.3.31-alpha.502a81.0<br/>@codecademy/gamut@32.18.8-alpha.502a81.0<br/>@codecademy/konamimojisplosion@0.2.31-alpha.502a81.0<br/>@codecademy/markdown-overrides@0.5.43-alpha.502a81.0<br/>@codecademy/styleguide@37.5.9-alpha.502a81.0<br/>@codecademy/variance@0.11.3-alpha.502a81.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://609dec92885a4d49d1372975--gamut-preview.netlify.app>https://609dec92885a4d49d1372975--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/609dec92885a4d49d1372975>Deploy Logs</a></p>

**Review by dreamwasp — APPROVED** [review]

File: packages/gamut-styles/package.json [hunk 1]
```diff
@@ -18,7 +18,7 @@
   },
   "dependencies": {
     "@codecademy/gamut-system": "^0.7.2",
-    "polished": "^4.1.2",
+    "polished": "^4.1.1",
```
Comment (JoshuaKGoldberg) [inline packages/gamut-styles/package.json:21]: 🤔 🤔 🤔 

Unintentional? 😄

**Review by JoshuaKGoldberg — APPROVED** [review]
Lol @ `boxShadows`. I honestly forgot we still had those floating around.

Very excellent, thanks for this!

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-icons@4.12.3-alpha.e10248.0<br/>@codecademy/gamut-illustrations@0.14.4-alpha.e10248.0<br/>@codecademy/gamut-kit@0.3.104-alpha.e10248.0<br/>@codecademy/gamut-labs@17.1.2-alpha.e10248.0<br/>@codecademy/gamut-patterns@0.1.4-alpha.e10248.0<br/>@codecademy/gamut-styles@10.0.1-alpha.e10248.0<br/>@codecademy/gamut-tests@2.3.31-alpha.e10248.0<br/>@codecademy/gamut@32.18.9-alpha.e10248.0<br/>@codecademy/konamimojisplosion@0.2.31-alpha.e10248.0<br/>@codecademy/markdown-overrides@0.5.43-alpha.e10248.0<br/>@codecademy/styleguide@38.1.2-alpha.e10248.0<br/>@codecademy/variance@0.11.3-alpha.e10248.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://60a2e2106ea5e33af7a9ac3a--gamut-preview.netlify.app>https://60a2e2106ea5e33af7a9ac3a--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/60a2e2106ea5e33af7a9ac3a>Deploy Logs</a></p>

**Pull request merged**